lib/StripeResource.js

```diff
@@ -282,15 +342,31 @@ StripeResource.prototype = {
         path: path,
       });
 
+      var requestRetries = numRetries || 0;
+
       req._requestEvent = requestEvent;
 
       req._requestStart = Date.now();
 
       self._stripe._emitter.emit('request', requestEvent);
 
       req.setTimeout(timeout, self._timeoutHandler(timeout, req, callback));
-      req.on('response', self._responseHandler(req, callback));
-      req.on('error', self._errorHandler(req, callback));
+
+      req.on('response', function(res) {
+        if (self._shouldRetry(res, requestRetries)) {
+          retryRequest(makeRequest, apiVersion, headers, requestRetries);
+        } else {
+          return self._responseHandler(req, callback)(res);
+        }
+      });
+
+      req.on('error', function(error) {
+        if (self._shouldRetry(null, requestRetries)) {
+          retryRequest(makeRequest, apiVersion, headers, requestRetries);
```

Comment:
we don't need to `return` this? (I'm not that familiar with `req.on('request'` et al)